test/externalTests/exttest/common.py

```diff
@@ -0,0 +1,336 @@
+                raise InvalidConfigError("Test directory not defined")
+
+            os.chdir(self.test_dir)
+            return fn(self, *args, **kwargs)
+
+        return f
+
+    @abstractmethod
+    def setup_environment(self, test_dir: Path):
+        pass
+
+    @abstractmethod
+    def clean(self):
+        pass
+
+    @abstractmethod
+    def compiler_settings(self, solc_version: str, presets: List[str]):
+        pass
+
+    @abstractmethod
+    def compile(self, solc_version: str, preset: str):
+        pass
+
+    @abstractmethod
+    def run_test(self, preset: str):
+        pass
+
+
+# Helper functions
+def compiler_settings(evm_version, via_ir="false", optimizer="false", yul="false") -> dict:
+    return {
+        "optimizer": {"enabled": optimizer, "details": {"yul": yul}},
+        "evmVersion": evm_version,
+        "viaIR": via_ir,
+    }
+
+
+def settings_from_preset(preset: str, evm_version: str) -> dict:
+    if preset not in AVAILABLE_PRESETS:
+        raise InvalidConfigError(
+            f'Preset "{preset}" not found.\n'
+            "Please select one or more of the available presets: " +
+            " ".join(AVAILABLE_PRESETS) + "\n"
+        )
+    switch = {
+        "legacy-no-optimize": compiler_settings(evm_version),
+        "ir-no-optimize": compiler_settings(evm_version, via_ir="true"),
+        "legacy-optimize-evm-only": compiler_settings(evm_version, optimizer="true"),
+        "ir-optimize-evm-only": compiler_settings(evm_version, via_ir="true", optimizer="true"),
+        "legacy-optimize-evm+yul": compiler_settings(evm_version, optimizer="true", yul="true"),
+        "ir-optimize-evm+yul":
+            compiler_settings(evm_version, via_ir="true", optimizer="true", yul="true"),
+    }
+    assert preset in switch
+    return switch[preset]
+
+
+def parse_command_line(description: str, args: List[str]):
+    arg_parser = ArgumentParser(description)
+    arg_parser.add_argument(
+        "solc_binary_type",
+        metavar="solc-binary-type",
+        type=str,
+        help="""Solidity compiler binary type""",
+        choices=["native", "solcjs"],
+    )
+    arg_parser.add_argument(
+        "solc_binary_path",
+        metavar="solc-binary-path",
+        type=str,
+        help="""Path to solc or soljson.js binary""",
+    )
+    return arg_parser.parse_args(args)
+
+
+def download_project(test_dir: Path, repo_url: str, ref_type: str = "branch", ref: str = "master"):
+    if ref_type not in ("commit", "branch", "tag"):
+        raise InvalidConfigError(f"Invalid git reference type: {ref_type}")
+
+    print(f"Cloning {ref_type} {ref} of {repo_url}...")
+    if ref_type == "commit":
+        os.mkdir(test_dir)
+        os.chdir(test_dir)
+        run_git_command(["git", "init"])
+        run_git_command(["git", "remote", "add", "origin", repo_url])
+        run_git_command(["git", "fetch", "--depth", "1", "origin", ref])
+        run_git_command(["git", "reset", "--hard", "FETCH_HEAD"])
```

Comment:
If you're not interested in capturing output it would be better to use `subprocess.run(command, check=True)` like you did for node. Using `run_git_command()` this way effectively silences the output.